File: app/channels/irrigation_status_channel.rb
# app/channels/irrigation_status_channel.rb
class IrrigationStatusChannel < ApplicationCable::Channel
  def subscribed
    stream_from "irrigation_status_\#{params[:sensor_id]}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/irrigation_status_channel.js
import consumer from "./consumer"



export const subscribeToIrrigation = (sensorId, updateUI) => {
  return consumer.subscriptions.create(
    { channel: "IrrigationStatusChannel", sensor_id: sensorId },
    {
      received(data) {
        updateUI(data)
      }
    }
  )
}
